textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java

```diff
@@ -66,23 +63,41 @@ public void open() throws Exception {
         try {
             positionIndex = 0;
             attributeIndex = 0;
-            operator.open();
-            dictionaryValue = dictionary.getNextValue();
+            dictionaryValue = predicate.getNextDictionaryValue();
 
-            // Java regex is used to detect word boundaries for TextField match.
-            // '\b' is used to match the beginning and end of the word.
-            regex = "\\b" + dictionaryValue.toLowerCase() + "\\b";
-            pattern = Pattern.compile(regex);
+            if (predicate.getSourceOperatorType() == DataConstants.SourceOperatorType.PHRASEOPERATOR) {

```

Comment:
I think since DictionaryMatcher is using KeywordMatcher (DictionaryMatcher has no direct access to IndexBasedSourceOperator), we have to explicitly define the functionality in DictionaryMatcher. But I'll check with @sandeepreddy602 once again regarding this.